docs/DeveloperGuide.adoc

```diff
@@ -153,94 +181,369 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+Step 5. The user executes `find task...` command to find a list of task given a keyword.
+The find command calls `Model#hasDashboard()`, which searches through the
+`UniqueDashboardList` for tasks containing the particular keyword. The `currentStatePointer`
+remains at the same location after the step.
 
-[NOTE]
-If the `currentStatePointer` is at index `addressBookStateList.size() - 1`, pointing to the latest address book state, then there are no undone address book states to restore. The `redo` command uses `Model#canRedoAddressBook()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the redo.
+Step 6. The user executes `list task` command to view all the tasks found in `UniqueDashboardList`.
+The list command calls `Model#getDashboardRecords()`, which returns the `ReadOnlyDashboard` that
+returns the `UniqueDashboardList`, iterating over all the task and displaying it on the
+homepage. The `currentStatePointer` remains at the same location after the step.
+
+The following sequence diagram shows how the add operation works:
+
+image::AddTaskSequenceDiagram.png[]
+
+*Aspect: How add/delete/edit/find/list executes*
+
+*Alternative 1 (current choice)*
+
+Sort the list whenever add/delete/edit is called.
+
+* Pros: Easy to implement.
+* Cons: May have performance issues in terms of time.
+
+*Alternative 2*
+
+Keep an instance of the earliest and latest date. If dates fall
+out of range from the two dates, sorting does not happen.
+
+* Pros: Faster runtime.
+* Cons: We must ensure that the implementation of each individual command are correct.
+
+*Aspect: Data structure to support the add/delete/edit/find/list commands*
+
+*Alternative 1 (current choice)*
+
+Use `Collections.sort` to sort by date.
+
+* Pros: Easy to implement.
+* Cons: Current implementations require a `LocalDate` object to be create for every
+`TaskDate` object.
+
+*Alternative 2*
+
+Use `LocalDate` to store dates instead of creating a custom class,
+`TaskDate`.
+
+* Pros: Since this is found in the Java API, we do not have to worry about invalid dates
+being keyed.
+* Cons: As the current `TaskDate` works fine, refactoring the code to `LocalDate` might
+take some time.
+// end::dashboard[]
+
+=== RecipeBook feature
+The current implementation of Diary consists of the following:
+
+* Each `Recipe` consists of a unique `name`
+* Each `Recipe` contains an Set of `Ingredient`
+* Each `Recipe` consists of a unique `Calories`
+* Each `Recipe` consists of a unique `Carbohydrates`
+* Each `Recipe` consists of a unique `Fats`
+* Each `Recipe` consists of a unique `Protein`
+* Each class has their respective getter methods
+
+The class diagram below gives an overview of the `Recipe` class.
+
+.Recipe Class Diagram
+image::RecipeClassDiagram.png[]
```

Comment:
Hi would it be better if the UML diagram has only either associations via the arrows or associations via the attributes?